latch_cli/centromere/utils.py

```diff
@@ -0,0 +1,223 @@
+            tls_verify = False
+        else:
+            tls_verify = tls_verify is not None
+
+        enable_tls = cert_path or tls_verify
+
+        dkr_client = None
+        try:
+            if not enable_tls:
+                dkr_client = docker.APIClient(host)
+            else:
+                if not cert_path:
+                    cert_path = os.path.join(os.path.expanduser("~"), ".docker")
+
+                tls_config = docker.tls.TLSConfig(
+                    client_cert=(
+                        os.path.join(cert_path, "cert.pem"),
+                        os.path.join(cert_path, "key.pem"),
+                    ),
+                    ca_cert=os.path.join(cert_path, "ca.pem"),
+                    verify=tls_verify,
+                )
+                dkr_client = docker.APIClient(host, tls=tls_config)
+
+        except docker.errors.DockerException as de:
+            raise OSError(
+                "Unable to establish a connection to Docker. Make sure that"
+                " Docker is running and properly configured before attempting"
+                " to register a workflow."
+            ) from de
+
+        return dkr_client
+
+    if ssh_host is not None:
+        try:
+            return docker.APIClient(ssh_host, use_ssh_client=True)
+        except docker.errors.DockerException as de:
+            raise OSError(
+                f"Unable to establish a connection to remote docker host {ssh_host}."
+            ) from de
+
+    environment = os.environ
+    host = environment.get("DOCKER_HOST")
+    if host is not None and host != "":
+        return _from_env()
+    else:
+        try:
+            # TODO: platform specific socket defaults
+            return docker.APIClient(base_url="unix://var/run/docker.sock")
+        except docker.errors.DockerException as de:
+            raise OSError(
+                "Docker is not running. Make sure that"
+                " Docker is running before attempting to register a workflow."
+            ) from de
+
+
+def construct_ssh_client(host_ip: str, username: str):
+
+    ssh = paramiko.SSHClient()
+    ssh.load_system_host_keys()
+    ssh.connect(host_ip, username=username)
+    return ssh
+
+
+class TmpDir:
+
+    """Represents a temporary directory that can be local or on a remote machine."""
+
+    def __init__(self, ssh_client=None, remote=False):
+
+        if remote and not ssh_client:
+            raise ValueError("Must provide an ssh client if remote is True.")
+
+        self.remote = remote
+        self.ssh_client = ssh_client
+        self._tempdir = None
+
+    def __enter__(self, *args):
+        return self.create(*args)
+
+    def __exit__(self, *args):
+        self.cleanup(*args)
+
+    def create(self, *args):
+        if not self.remote:
+            self._tempdir = tempfile.TemporaryDirectory()
+            return Path(self._tempdir.name).resolve()
+        else:
+            td = "".join(
+                random.choice(
```

Comment:
```suggestion
                random.choices(
```
https://docs.python.org/3/library/random.html#random.choices